semantic-core/src/Analysis/Analysis.hs

```diff
@@ -3,25 +3,24 @@ module Analysis.Analysis
 ( Analysis(..)
 ) where
 
-import Core.Name
 import Data.Text (Text)
 
 -- | A record of functions necessary to perform analysis.
 --
 -- This is intended to be replaced with a selection of algebraic effects providing these interfaces and carriers providing reusable implementations.
-data Analysis term address value m = Analysis
-  { alloc     :: Name -> m address
-  , bind      :: forall a . Name -> address -> m a -> m a
-  , lookupEnv :: Name -> m (Maybe address)
+data Analysis term name address value m = Analysis
+  { alloc     :: name -> m address
+  , bind      :: forall a . name -> address -> m a -> m a
+  , lookupEnv :: name -> m (Maybe address)
   , deref     :: address -> m (Maybe value)
   , assign    :: address -> value -> m ()
-  , abstract  :: (term -> m value) -> Name -> term -> m value
-  , apply     :: (term -> m value) -> value -> value -> m value
+  , abstract  :: (term name -> m value) -> name -> term name -> m value
+  , apply     :: (term name -> m value) -> value -> value -> m value
   , unit      :: m value
   , bool      :: Bool -> m value
   , asBool    :: value -> m Bool
   , string    :: Text -> m value
   , asString  :: value -> m Text
-  , record    :: [(Name, value)] -> m value
-  , (...)     :: address -> Name -> m (Maybe address)
+  , record    :: [(name, value)] -> m value
+  , (...)     :: address -> name -> m (Maybe address)
   }
```

Comment:
Two changes worth noting here:

1. We now parameterize `Analysis` by the type of names.

2. The `term` parameter is now applied to `name`, i.e. it’s `* -> *`. This is an assumption of the `Concrete` analysis (which it uses to close over the free variables in function bodies), and I’d like to avoid having analyses with different assumptions as to the shape of that type.